modules/lifestreetBidAdapter.js

```diff
@@ -0,0 +1,143 @@
+import { isInteger } from '../src/utils.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { BANNER, VIDEO } from '../src/mediaTypes.js';
+
+const BIDDER_CODE = 'lifestreet';
+const ADAPTER_VERSION = '$prebid.version$';
+
+const urlTemplate = template`https://ads.lfstmedia.com/gate/${'adapter'}/${'slot'}?adkey=${'adkey'}&ad_size=${'ad_size'}&__location=${'location'}&__referrer=${'referrer'}&__wn=${'wn'}&__sf=${'sf'}&__fif=${'fif'}&__if=${'if'}&__stamp=${'stamp'}&__pp=1&__hb=1&_prebid_json=1&__gz=1&deferred_format=vast_2_0,vast_3_0&__hbver=${'hbver'}`;
+
+/**
+ * A helper function for template to generate string from boolean
+ */
+function boolToString(value) {
+  return value ? '1' : '0';
+}
+
+/**
+ * A helper function to form URL from the template
+ */
+function template(strings, ...keys) {
+  return function(...values) {
+    let dict = values[values.length - 1] || {};
+    let result = [strings[0]];
+    keys.forEach(function(key, i) {
+      let value = isInteger(key) ? values[key] : dict[key];
+      result.push(value, strings[i + 1]);
+    });
+    return result.join('');
+  };
+}
+
+/**
+ * Creates a bid requests for a given bid.
+ *
+ * @param {BidRequest} bid The bid params to use for formatting a request
+ */
+function formatBidRequest(bid, bidderRequest = {}) {
+  const {params} = bid;
+  const {referer} = (bidderRequest.refererInfo || {});
```

Comment:
FYI. `refererInfo` was updated/refactored in Prebid 7 with #8450 . Not sure if you are expecting refererInfo to act as previous versions.